app/javascript/components/misc/CodeMirror.tsx

```diff
@@ -71,31 +71,32 @@ export const CodeMirror = ({
       return
     }
 
-    const view = new EditorView({
-      state: EditorState.create({
-        doc: value,
-        extensions: [
-          basicSetup,
-          languageCompartment(language),
-          //javascript(),
-          tabCaptureCompartment.of(
-            keymap.of(isTabCaptured ? [defaultTabBinding] : [])
-          ),
-          EditorState.tabSize.of(tabSize),
-          indentUnit.of(useSoftTabs ? '  ' : '	'),
-          wrapCompartment.of(wrap ? EditorView.lineWrapping : []),
-          themeCompartment.of(
-            EditorView.theme({}, { dark: theme === Themes.DARK })
-          ),
-          keymap.of(commands),
-        ],
-      }),
-      parent: textarea,
+    importLanguageCompartment(language).then((languageCompartment) => {
```

Comment:
@kntsoriano I haven't spent a lot of time on this, but my code is not working. This is most likely due to the fact that the language compartment is now retrieved asynchronously. I assume this means that this `useEffect` call is returning _before_ the language compartment has been retrieved, which might mess with subsequent `useEffect` calls. I'd be happy to look into this a bit more, but if you feel like you can solve this quickly, be my guest.